src/target/riscv/riscv-013.c

```diff
@@ -592,6 +496,7 @@ static program_t *program_new(void)
 		program->write = false;
 		program->regno = 0x1000;
 	}
+	program_add32(program, fence_i());
```

Comment:
Do you need this? The Debug ROM has a fence.i in it before executing the program buffer